### PR TITLE
SignUp Controller added and Tested

### PR DESCRIPTION
**Pull request by DeepakKumar03**
Added Sign Up controller, tested and formated the code.

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            userEntity.setUsername(signupUserRequest.getUserName());
+            userEntity.setEmail(signupUserRequest.getEmailAddress());
+            userEntity.setPassword(signupUserRequest.getPassword());
+            userEntity.setCountry(signupUserRequest.getCountry());
+            userEntity.setAboutme(signupUserRequest.getAboutMe());
+            userEntity.setDob(signupUserRequest.getDob());
+            userEntity.setContactnumber(signupUserRequest.getContactNumber());
+            userEntity.setRole("nonadmin");
+            final UserEntity createdUserEntity = signupBusinessService.signup(userEntity);
+            com.upgrad.quora.api.model.SignupUserResponse userResponse = new com.upgrad.quora.api.model.SignupUserResponse().id(createdUserEntity.getUuid()).status("USER SUCCESSFULLY REGISTERED");
```
Comment (simpleprahalad) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java]: Only SignupUserReponse should be good rnough.

File: quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java [hunk 2]
```diff
@@ -12,17 +13,23 @@
...
-  }
 
-  @ExceptionHandler(AuthorizationFailedException.class)
-  public ResponseEntity<ErrorResponse> authorizationFailedException(AuthorizationFailedException exe, WebRequest request){
-    return new ResponseEntity<ErrorResponse>(
-        new ErrorResponse().code(exe.getCode()).message(exe.getErrorMessage()), HttpStatus.UNAUTHORIZED
-    );
-  }
+    @ExceptionHandler(SignUpRestrictedException.class)
+    public ResponseEntity<ErrorResponse> signUpRestrictedException(SignUpRestrictedException ex, WebRequest request) {
```
Comment (simpleprahalad) [inline quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java]: Change the variable name to "exe" to maintain uniformity across code.

File: quora-service/src/main/java/com/upgrad/quora/service/business/SignupBusinessService.java [hunk 3]
```diff
@@ -19,12 +19,20 @@
...
         userEntity.setSalt(encryptedText[0]);
         userEntity.setPassword(encryptedText[1]);
         return userDao.createUser(userEntity);
     }
+
+    public UserEntity getUser(final String userName) {
+        return userDao.getUserByUserName(userName);
+    }
+
+    public UserEntity getEmail(final String email) {
```
Comment (simpleprahalad) [inline quora-service/src/main/java/com/upgrad/quora/service/business/SignupBusinessService.java]: Name of the method is incorrect. I think better name should be getUserbyEmail(). Please recheck

**Review by simpleprahalad — CHANGES_REQUESTED** [review]
Overall changes look fine. Please make the changes requested.

**Review by DeepakKumar03 — COMMENTED** [review]
Reviewed the comment and made modification as requested.

**Pull request closed without merging**